packages/react-router/modules/RouterContext.js

```diff
@@ -1,9 +1,12 @@
 // TODO: Replace with React.createContext once we can assume React 16+
 import createContext from "create-react-context";
 
-const context = createContext();
-
-context.Provider.displayName = "Router.Provider";
-context.Consumer.displayName = "Router.Consumer";
+const createNamedContext = name => {
```

Comment:
this avoids static property assignments by hiding it in a tree-shakeable function call (this was preventing tree shaking for simialr reason as `.defaultProps`)

Reply:
Sounds good 👍